src/shared_gui_components/BuildingComponentDialogCentralWidget.cpp

```diff
@@ -190,16 +217,17 @@ void BuildingComponentDialogCentralWidget::setTid(const std::string& filterType,
     delete comp;
   }
 
-  RemoteBCL remoteBCL;
-  remoteBCL.setTimeOutSeconds(m_timeoutSeconds);
-  std::vector<BCLSearchResult> responses;
-  if (filterType == "components") {
-    responses = remoteBCL.searchComponentLibrary(searchString.toStdString(), tid, pageIdx);
-  } else if (filterType == "measures") {
-    responses = remoteBCL.searchMeasureLibrary(searchString.toStdString(), tid, pageIdx);
+  if (pageIdx == 0 || m_allResponses.empty()) {
```

Comment:
This would be a bug if you changed `tid` or `searchString` but then called again with `pageIdx=0`.  You need something like `m_lastSearchString =  filterType + tid + searchString`, then clear `m_allResponses` when that last string cache key changes

Reply:
This should be addressed by the lines


```
  if (newKey != currentKey) {
    openstudio::RemoteBCL remoteBCL;
    m_allResponses = fetchAndSortResponses(filterType, tid, searchString, remoteBCL);
  }
```